src/lib.rs

```diff
@@ -36,6 +34,27 @@ extern crate stable_deref_trait;
 #[cfg(feature = "unsize")]
 extern crate unsize;
 
+/// Calculates the offset of the specified field from the start of the named struct.
+macro_rules! offset_of {
+    ($ty: path, $field: tt) => {{
+        let $ty { $field: _, .. };
```

Comment:
can we wrap this in a block that evaluates to a const so that it's forced to be const?

Also it's not clear to me what this first line is doing

Reply:
See the latest commit

Reply:
is the pointer subtraction not const?

Reply:
Yes. Raw pointers may in different address spaces (more specifically, called [pointer provenances](https://rust-lang.github.io/unsafe-code-guidelines/glossary.html#pointer-provenance)) during const evaluation. Calculating offset requires special handling.